lwb/metalang/spt/spt/src/main/java/mb/spt/expectation/RunExpectation.java

```diff
@@ -0,0 +1,139 @@
+import mb.common.message.Severity;
+import mb.common.option.Option;
+import mb.common.region.Region;
+import mb.common.result.Result;
+import mb.common.util.ListView;
+import mb.pie.api.ExecContext;
+import mb.pie.api.Session;
+import mb.pie.api.exec.CancelToken;
+import mb.resource.ResourceKey;
+import mb.spoofax.core.language.LanguageInstance;
+import mb.spt.api.analyze.RunArgument;
+import mb.spt.api.analyze.TestableAnalysis;
+import mb.spt.fromterm.InvalidAstShapeException;
+import mb.spt.lut.LanguageUnderTestProvider;
+import mb.spt.model.LanguageUnderTest;
+import mb.spt.model.SelectionReference;
+import mb.spt.model.TestCase;
+import mb.spt.model.TestExpectation;
+import mb.stratego.common.StrategoException;
+import org.spoofax.interpreter.terms.IStrategoAppl;
+import org.spoofax.interpreter.terms.IStrategoInt;
+import org.spoofax.interpreter.terms.IStrategoString;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+import org.spoofax.terms.util.TermUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RunExpectation implements TestExpectation {
+    private final String strategy;
+    private final Option<ListView<IStrategoAppl>> arguments;
+    private final Option<SelectionReference> selectionReference;
+    private final Region sourceRegion;
+    private final boolean expectFailure;
+
+    public RunExpectation(String strategy, Option<ListView<IStrategoAppl>> arguments, Option<SelectionReference> selectionReference, Region sourceRegion, boolean expectFailure) {
+        this.strategy = strategy;
+        this.arguments = arguments;
+        this.selectionReference = selectionReference;
+        this.sourceRegion = sourceRegion;
+        this.expectFailure = expectFailure;
+    }
+
+    protected void checkAst(
+        IStrategoTerm ast,
+        KeyedMessagesBuilder messagesBuilder,
+        TestCase testCase,
+        LanguageUnderTest languageUnderTest,
+        Session languageUnderTestSession, LanguageUnderTestProvider languageUnderTestProvider,
+        ExecContext context,
+        Region sourceRegion
+    ) throws InterruptedException {}
+
+    @Override
+    public KeyedMessages evaluate(
+        TestCase testCase,
+        LanguageUnderTest languageUnderTest,
+        Session languageUnderTestSession,
+        LanguageUnderTestProvider languageUnderTestProvider,
+        ExecContext context,
+        CancelToken cancel
+    ) throws InterruptedException {
+        final ResourceKey file = testCase.testSuiteFile;
+        final KeyedMessagesBuilder messagesBuilder = new KeyedMessagesBuilder();
+        final LanguageInstance languageInstance = languageUnderTest.getLanguageComponent().getLanguageInstance();
+        if(!(languageInstance instanceof TestableAnalysis)) {
+            messagesBuilder.addMessage("Cannot evaluate parse to ATerm expectation because language instance '" + languageInstance + "' does not implement TestableAnalysis", Severity.Error, file, sourceRegion);
```

Comment:
Adjust error message to use "run expectation" instead of "parse to ATerm expectation"